sstable.go

```diff
@@ -116,7 +132,62 @@ func (s *SSTable) readIndex() (*Index, error) {
 	return NewIndex(index), nil
 }
 
+func (s *SSTable) bloomFilterOffset() (uint64, error) {
+	fileSize, err := s.file.Stat()
+	if err != nil {
+		return 0, err
+	}
+
+	bloomFilterSize, err := s.bloomFilterSize()
+	if err != nil {
+		return 0, err
+	}
+
+	// 전체 파일 크기
+	totalSize := uint64(fileSize.Size())
+
+	// bloomFilterOffset 계산
+	// 전체 크기 - (footer 크기 + bloomFilter 크기)
+	bloomFilterOffset := totalSize - (16 + bloomFilterSize)
+
+	return bloomFilterOffset, nil
+}
+
+func (s *SSTable) bloomFilterSize() (uint64, error) {
+	footer, err := s.readFooter()
+	if err != nil {
+		return 0, err
+	}
+	return binary.LittleEndian.Uint64(footer[8:]), nil
+}
+
+func (s *SSTable) readBloomFilter() (*BloomFilter, error) {
+	bloomFilterOffset, err := s.bloomFilterOffset()
+	if err != nil {
+		return nil, err
+	}
+	bloomFilterSize, err := s.bloomFilterSize()
+	if err != nil {
+		return nil, err
+	}
+	bloomFilter := make([]byte, bloomFilterSize)
+	_, err = s.file.ReadAt(bloomFilter, int64(bloomFilterOffset))
```

Comment:
**Optimizations:**
The `readBloomFilter` function reads the entire Bloom filter into memory. If the Bloom filter is large, this could be inefficient. Consider streaming the data or using a more memory-efficient approach.

```go
func (s *SSTable) readBloomFilter() (*BloomFilter, error) {
    bloomFilterOffset, err := s.bloomFilterOffset()
    if err != nil {
        return nil, err
    }
    bloomFilterSize, err := s.bloomFilterSize()
    if err != nil {
        return nil, err
    }
    bloomFilter := make([]byte, bloomFilterSize)
    _, err = s.file.ReadAt(bloomFilter, int64(bloomFilterOffset))
    if err != nil {
        return nil, err
    }
    return LoadBloomFilter(bloomFilter)
}
```